From c003d63ced416afd2998547a13af85e68108e75e Mon Sep 17 00:00:00 2001
From: Alexandre d'Alton <alexandre.d'alton@intel.com>
Date: Wed, 7 Aug 2013 14:41:38 +0300
Subject: [PORT FROM MAIN] Telephony: Fix issue in getting operator numeric code

BZ: 118747 122916

Upon IMSI ready, call to getOperatorNumberic function returns
NULL.

mncLength is assigned to UNINITIALIZED in the resetRecords
function. When the IMSI is read successfully, mncLength is
still not set to other value. So, call to getOperatorNumeric
will result in NULL as the function checks the mncLength
validity.

Reading of the IMSI is triggered once the EF_AD is read in
order to allow getOperatorNumeric to be successful after the
IMSI ready notification.

Change-Id: Ic73e9c1f9a8d765211e3653f4d6eec61791e5ca6
Orig-Change-Id: I912b86f088ac16cbfb0c86a4aee28ae9dd33e2a6
Category: device enablement
Domain: Telephony-Modem-Common
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Alexandre d'Alton <alexandre.d'alton@intel.com>
Signed-off-by: Lionel Ulmer <lionel.ulmer@intel.com>
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
---
 .../internal/telephony/uicc/SIMRecords.java        |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/java/com/android/internal/telephony/uicc/SIMRecords.java b/src/java/com/android/internal/telephony/uicc/SIMRecords.java
index 72b4d56..f532531 100644
--- a/src/java/com/android/internal/telephony/uicc/SIMRecords.java
+++ b/src/java/com/android/internal/telephony/uicc/SIMRecords.java
@@ -897,6 +897,9 @@ public class SIMRecords extends IccRecords {
                         MccTable.updateMccMncConfiguration(mContext,
                                 mImsi.substring(0, 3 + mMncLength));
                     }
+
+                    mCi.getIMSIForApp(mParentApp.getAid(), obtainMessage(EVENT_GET_IMSI_DONE));
+                    mRecordsToLoad++;
                 }
             break;
 
@@ -1384,7 +1387,7 @@ public class SIMRecords extends IccRecords {
 
         if (DBG) log("fetchSimRecords " + mRecordsToLoad);
 
-        mCi.getIMSIForApp(mParentApp.getAid(), obtainMessage(EVENT_GET_IMSI_DONE));
+        mFh.loadEFTransparent(EF_AD, obtainMessage(EVENT_GET_AD_DONE));
         mRecordsToLoad++;
 
         mFh.loadEFTransparent(EF_ICCID, obtainMessage(EVENT_GET_ICCID_DONE));
@@ -1400,9 +1403,6 @@ public class SIMRecords extends IccRecords {
         mFh.loadEFLinearFixed(EF_MBI, 1, obtainMessage(EVENT_GET_MBI_DONE));
         mRecordsToLoad++;
 
-        mFh.loadEFTransparent(EF_AD, obtainMessage(EVENT_GET_AD_DONE));
-        mRecordsToLoad++;
-
         // Record number is subscriber profile
         mFh.loadEFLinearFixed(EF_MWIS, 1, obtainMessage(EVENT_GET_MWIS_DONE));
         mRecordsToLoad++;
-- 
1.7.4.1

